src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportOrchestratorTask.cs

```diff
@@ -0,0 +1,505 @@
+
+                    ImportTaskErrorResult errorResult = new ImportTaskErrorResult();
+                    errorResult.HttpStatusCode = HttpStatusCode.InternalServerError;
+                    errorResult.ErrorMessage = ex.Message;
+
+                    throw new RetriableTaskException(JsonConvert.SerializeObject(errorResult));
+                }
+            }
+
+            if (taskResultData == null) // No exception
+            {
+                taskResultData = new TaskResultData(TaskResult.Success, JsonConvert.SerializeObject(_orchestratorTaskContext.ImportResult));
+            }
+
+            return taskResultData;
+        }
+
+        public void Cancel()
+        {
+            if (!_cancellationTokenSource.IsCancellationRequested)
+            {
+                _cancellationTokenSource?.Cancel();
+            }
+        }
+
+        private static long CalculateResourceNumberByResourceSize(long blobSizeInBytes, long resourceCountPerBytes)
+        {
+            return Math.Max((blobSizeInBytes / resourceCountPerBytes) + 1, 10000L);
+        }
+
+        private async Task ValidateResourcesAsync(CancellationToken cancellationToken)
+        {
+            foreach (var input in _orchestratorInputData.Input)
+            {
+                Dictionary<string, object> properties = await _integrationDataStoreClient.GetPropertiesAsync(input.Url, cancellationToken);
+                if (!string.IsNullOrEmpty(input.Etag))
+                {
+                    if (!input.Etag.Equals(properties[IntegrationDataStoreClientConstants.BlobPropertyETag]))
+                    {
+                        throw new ImportFileEtagNotMatchException(string.Format("Input file Etag not match. {0}", input.Url));
+                    }
+                }
+            }
+        }
+
+        private async Task UpdateProgressAsync(ImportOrchestratorTaskContext context, CancellationToken cancellationToken)
+        {
+            await _contextUpdater.UpdateContextAsync(JsonConvert.SerializeObject(context), cancellationToken);
+        }
+
+        private async Task<Dictionary<Uri, TaskInfo>> GenerateSubTaskRecordsAsync(CancellationToken cancellationToken)
+        {
+            Dictionary<Uri, TaskInfo> result = new Dictionary<Uri, TaskInfo>();
+
+            long beginSequenceId = _sequenceIdGenerator.GetCurrentSequenceId();
+
+            foreach (var input in _orchestratorInputData.Input)
+            {
+                string taskId = Guid.NewGuid().ToString("N");
+
+                Dictionary<string, object> properties = await _integrationDataStoreClient.GetPropertiesAsync(input.Url, cancellationToken);
+                long blobSizeInBytes = (long)properties[IntegrationDataStoreClientConstants.BlobPropertyLength];
+                long estimatedResourceNumber = CalculateResourceNumberByResourceSize(blobSizeInBytes, DefaultResourceSizePerByte);
+                long endSequenceId = beginSequenceId + estimatedResourceNumber;
```

Comment:
What are the begin and end sequence id used for? It is just to give the user an estimate of how many resources were ingested?

Reply:
It looks like this is the block of Ids assigned to this input file. So the expectation is that DefaultResourceSizePerByte is smaller than the average resource size in the file?

Reply:
For our scenario of SQL, it is ResourceSurrogateId. 

For DefaultResourceSizePerByte, it is much small than resource private const long DefaultResourceSizePerByte = 64;